modules/calib3d/src/usac/local_optimization.cpp

```diff
@@ -345,293 +332,272 @@ Ptr<InnerIterativeLocalOptimization> InnerIterativeLocalOptimization::create
             lo_inner_iterations_, lo_iter_max_iterations_, threshold_multiplier_);
 }
 
-class SigmaConsensusImpl : public SigmaConsensus {
+class SimpleLocalOptimizationImpl : public SimpleLocalOptimization {
 private:
-    const Ptr<Estimator> estimator;
     const Ptr<Quality> quality;
     const Ptr<Error> error;
-    const Ptr<ModelVerifier> verifier;
-    const GammaValues& gamma_generator;
-    // The degrees of freedom of the data from which the model is estimated.
-    // E.g., for models coming from point correspondences (x1,y1,x2,y2), it is 4.
-    const int degrees_of_freedom;
-    // A 0.99 quantile of the Chi^2-distribution to convert sigma values to residuals
-    const double k;
-    // Calculating (DoF - 1) / 2 which will be used for the estimation and,
-    // due to being constant, it is better to calculate it a priori.
-    double dof_minus_one_per_two;
-    const double C;
-    // The size of a minimal sample used for the estimation
-    const int sample_size;
-    // Calculating 2^(DoF - 1) which will be used for the estimation and,
-    // due to being constant, it is better to calculate it a priori.
-    double two_ad_dof;
-    // Calculating C * 2^(DoF - 1) which will be used for the estimation and,
-    // due to being constant, it is better to calculate it a priori.
-    double C_times_two_ad_dof;
-    // Calculating the gamma value of (DoF - 1) / 2 which will be used for the estimation and,
-    // due to being constant, it is better to calculate it a priori.
-    double squared_sigma_max_2, one_over_sigma;
-    // Calculating the upper incomplete gamma value of (DoF - 1) / 2 with k^2 / 2.
-    const double gamma_k;
-    // Calculating the lower incomplete gamma value of (DoF - 1) / 2 which will be used for the estimation and,
-    // due to being constant, it is better to calculate it a priori.
-    double max_sigma_sqr;
-    const int points_size, number_of_irwls_iters;
-    const double maximum_threshold, max_sigma;
-
-    std::vector<double> sqr_residuals, sigma_weights;
-    std::vector<int> sqr_residuals_idxs;
-    // Models fit by weighted least-squares fitting
-    std::vector<Mat> sigma_models;
-    // Points used in the weighted least-squares fitting
-    std::vector<int> sigma_inliers;
-    // Weights used in the the weighted least-squares fitting
-    int max_lo_sample_size, stored_gamma_number_min1;
-    double scale_of_stored_gammas;
-    RNG rng;
-    const std::vector<double> &stored_gamma_values;
+    const Ptr<NonMinimalSolver> estimator;
+    const Ptr<Termination> termination;
+    const Ptr<RandomGenerator> random_generator;
+    const Ptr<WeightFunction> weight_fnc;
+    Ptr<RandomGenerator> random_generator2;
```

Comment:
It's not clear from name why you need the second generator.

Reply:
I added description, it is more efficient to have a separate one.